src/Image/Operation/ToJpg.php

```diff
@@ -67,6 +67,10 @@ public function run($load_filename, $save_filename)
 
         $input = $imagecreate_function($load_filename);
 
+        if (!$input) {
```

Comment:
With PHPStan being more strict as level increase, I changed my habits regarding those conditions and now check the expected value (this is also advised by the mighty @szepeviktor [here](https://github.com/szepeviktor/very-clean-code#-write-positive-conditions)).

```php
if (!$input instanceof GdImage) {
    return false;
}

```

Reply:
> I am just a person.

I doubt that.

Reply:
Hi @nlemoine,

Thanks for the suggestion. Just patched that.

Can you clear something up for me as well? From what I can see in the PHP docs, imagecreatefrompng() for example is already typed. Is the `instanceof` still needed in that case?

```php
imagecreatefrompng(string $filename): GdImage|false
```

Reply:
One can go three ways.
1. Type juggling to boolean `!$input` - the worst way 🔴 
2. Weak implicit check `$input === false` 🟠 
3. Strong explicit `!$input instanceof GdImage` 🟢

Reply:
@nlemoine Seems PHP 7,4 is failing because the GdImage was only added since PHP 8.0. See changelog: https://www.php.net/manual/en/function.imagecreatefrompng.php. How to proceed?

Reply:
> How to proceed?

Oh noooooooooooo!
Then we need 🟠

Reply:
Drop php 7.4? 😅

I'd say weak implicit check. This might be overkill to check for GdImage existence and adapt the condition.